returns/interfaces/filterable.py

```diff
@@ -0,0 +1,54 @@
+from abc import abstractmethod
+from typing import Callable, NoReturn, TypeVar
+
+from returns.interfaces.specific.maybe import MaybeLikeN
+from returns.primitives.hkt import Kind1
+
+_FirstType = TypeVar('_FirstType')
+_SecondType = TypeVar('_SecondType')
+_ThirdType = TypeVar('_ThirdType')
+
+_FilterableType = TypeVar('_FilterableType', bound='FilterableN')
+
+
+class FilterableN(MaybeLikeN[_FirstType, _SecondType, _ThirdType]):
+    """
+    Represents container that can apply filter over inner value.
+
+    There are no aliases or ``FilterableN` for ``Filterable`` interface.
+    Because it always uses one type.
+
+    Not all types can be ``Filterable`` because we require
+    a possibility to access internal value and to model a case,
+    where the predicate is false
+
+    .. code:: python
+
+        >>> from returns.maybe import Nothing, Some
+        >>> from returns.pointfree import filter_
+
+        >>> def example(argument: int) -> bool:
```

Comment:
what would you think of naming the example function? Naming the function `is_even` makes the example clearer in my mind